cdv/cmds/sim_utils.py

```diff
@@ -0,0 +1,525 @@
+import asyncio
+import os
+from decimal import Decimal
+from pathlib import Path
+from random import randint
+from typing import Any, Callable, Dict, List, Optional
+
+from aiohttp import ClientConnectorError
+from blspy import PrivateKey
+from chia.consensus.coinbase import create_puzzlehash_for_pk
+from chia.simulator.simulator_full_node_rpc_client import SimulatorFullNodeRpcClient
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_record import CoinRecord
+from chia.util.bech32m import decode_puzzle_hash, encode_puzzle_hash
+from chia.util.config import load_config, save_config
+from chia.util.ints import uint32
+from chia.util.keychain import Keychain, bytes_to_mnemonic
+from chia.wallet.derive_keys import (
+    master_sk_to_farmer_sk,
+    master_sk_to_pool_sk,
+    master_sk_to_wallet_sk,
+    master_sk_to_wallet_sk_unhardened,
+)
+
+SIMULATOR_ROOT_PATH = Path(os.path.expanduser(os.getenv("CHIA_SIMULATOR_ROOT", "~/.chia/simulator"))).resolve()
+
+
+# based on execute_with_node
+async def execute_with_simulator(
+    rpc_port: Optional[int], root_path: Path, function: Callable, catch_errors: bool = True, *args
+) -> Any:
+    import traceback
+
+    from chia.util.config import load_config
+    from chia.util.ints import uint16

```

Comment:
Is there a reason you're doing imports inside of the function?  These don't seem like things that matter to be optional so it seems like you're just slowing down the runtime by doing this.

Reply:
ill move them out